docs/docs/aleph_zero/sdk.md

```diff
@@ -0,0 +1,413 @@
+
+```typescript
+const api = await initPolkadotApi(Network.Main) // initialize api, use enum to specify the network
+
+// initialize account, you can use your own wallet by pasting mnemonic phase
+const keyring = new Keyring({ type: 'sr25519' })
+const account = await keyring.addFromUri('//Alice')
+```
+
+### Initialize DEX and tokens
+
+:::note Why "n" is at the end of every number
+Notice how we use "n" at the end of every number. "n" indicates that specified value is a BigInt, number with higher precision. Read more about BigInt here: [BigInt MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt).
+:::
+
+:::info Creating types
+You can create custom decimals with `toDecimal` syntax. First argument is an integer and second argument is a number of zeros. For example `toFee(3n, 2n)` will produce decimal equal to `0.03`. Read more about types here: [Types](types.md).
+:::
+
+:::warning Token sorting
+Tokens are sorted alphabetically when pool key is created, so make sure that you swap tokens in correct direction. Read more about pool keys here: [PoolKey](storage#poolkey).
+:::
+
+```typescript
+// --snip--
+
+// load invariant contract
+const invariant = await Invariant.load(api, Network.Local, INVARIANT_ADDRESS)
+
+// load token contract
+const psp22 = await PSP22.load(api, Network.Local, TOKEN0_ADDRESS)
+```
+
+### Create pool
+
+```typescript
+// --snip--
+
+// set fee tier, make sure that fee tier with specified parameters exists
+const feeTier = newFeeTier(toPercentage(1n, 2n), 1n) // fee: 0.01 = 1%, tick spacing: 1
+
+// set initial price of the pool, we set it to 1.00
+const initSqrtPrice = toSqrtPrice(1n, 0n)
+
+await invariant.createPool(account, TOKEN0_ADDRESS, TOKEN1_ADDRESS, feeTier, initSqrtPrice, 0n)
+```
+
+### Open position
+
+```typescript
+// token y has 12 decimals and we want to add 8 actual tokens to our position
+const tokenYAmount = 8n * 10n ** 12n
+
+// set lower and upper tick, we want to create position in range [-10, 10]
+const lowerTick = -10n
+const upperTick = 10n
+
+const liquidityResult = getLiquidityByX(tokenYAmount, lowerTick, upperTick, initSqrtPrice, true)
+const tokenXAmount = liquidityResult.amount
+const positionLiquidity = liquidityResult.l
+
+// print amount of token x and y we need to give to open position based on parameteres we passed
+console.log(tokenYAmount, tokenXAmount)
+
+// approve transfers of both tokens
+await psp22.approve(
+  account,
+  invariant.contract.address.toString(),
+  isTokenX(TOKEN0_ADDRESS, TOKEN1_ADDRESS) ? tokenYAmount : tokenXAmount
+)
+await psp22.setContractAddress(TOKEN1_ADDRESS)
+await psp22.approve(
+  account,
+  invariant.contract.address.toString(),
+  isTokenX(TOKEN0_ADDRESS, TOKEN1_ADDRESS) ? tokenXAmount : tokenYAmount
+)
+await psp22.setContractAddress(TOKEN0_ADDRESS)
+
+// open up position
+await invariant.createPosition(
+  account,
+  poolKey,
+  lowerTick,
+  upperTick,
+  positionLiquidity,
+  initSqrtPrice,
```

Comment:
There is a need to update the interface of `createPosition`: remove `targetSqrtPriceLower` and `targetSqrtPriceUpper`; instead, pass `spotSqrtPrice: SqrtPrice` and `sliappgeTolerance: Percentage`